spring-xd-module/src/main/java/org/springframework/xd/module/core/JavaConfiguredModule.java

```diff
@@ -62,8 +57,10 @@ protected void configureModuleApplicationContext(SimpleModuleDefinition moduleDe
 	}
 
 	public static String[] basePackages(SimpleModuleDefinition moduleDefinition, ClassLoader moduleClassLoader) {
-		Resource propertiesFile = ModuleUtils.locateModuleResource(moduleDefinition, moduleClassLoader,".properties");
-		Assert.notNull(propertiesFile, "required module properties not found.");
+		Resource propertiesFile = ModuleUtils.locateModuleResource(moduleDefinition, moduleClassLoader, ".properties");
+		if (propertiesFile == null) {
+			return new String[0];
```

Comment:
From the implementation of `scan()` downstream, seems like we'd better fail fast:
```java
	/**
	 * Perform a scan within the specified base packages.
	 * <p>Note that {@link #refresh()} must be called in order for the context
	 * to fully process the new classes.
	 * @param basePackages the packages to check for annotated classes
	 * @see #register(Class...)
	 * @see #refresh()
	 */
	public void scan(String... basePackages) {
		Assert.notEmpty(basePackages, "At least one base package must be specified");
		this.scanner.scan(basePackages);
	}
```